### DISCUSSION TITLE
Developing custom applications

### DISCUSSION THREAD

**Discussion by zia-javanbakht** [idaholab/moose#27793] [Q&A Modules: Solid mechanics]
I am trying to set up my own FE formulation using MOOSE for a sandwich panel.

I have done the tutorial for developing custom application. I have gone through shell, beam, and truss kernel and material source files. Nevertheless, it is very confusing. 

As a beginner user, I want to understand how to formulate my own application. For instance, I want to write a 2D truss application to practice MOOSE systems. Is there any way to learn this in a straightforward way? My goal is to learn and go back to my original sandwich panel problem.

I would appreciate any recommendations.

**Comment by GiudGiud** [comment]
@cbolisetti any recommendations ?

Working with custom (truss) elements is not beginner use of the software. It's a little advanced. Those Truss objects you found are going to be your best examples

**Reply by zia-javanbakht** [reply]
Thank you very much for your prompt response.

There are some confusing parts like useVectorTag(). I said truss since I thought that it would be trivial. :) Any recommendations is welcome. Many thanks.

**Reply by cbolisetti** [reply]
I don't think I understand the application here. Are you trying to develop a material model for sandwich panels or an app? If it's an app, I am a bit confused about why you'd like to create an app for one problem? 

And @GiudGiud mentioned, it might be a good idea to first use the code to solve your problem and then decide if you need an app (and what you'd like to put in that app).

**Reply by GiudGiud** [reply]
If you plan to develop, creating an app is a good idea. It's the recommended way.

**Reply by zia-javanbakht** [reply]
Yes, I want to develop an app for the sandwich panel. I have a formulation for it with different DoFs on each node, see [draft](https://www.researchgate.net/publication/333236990_On_Thermal_Strains_and_Residual_Stresses_in_the_Linear_Theory_of_Anti-Sandwiches).

Yes, I need to develop an app. I understand that this might not be efficient. I don't see any other way though.

Any thoughts?

**Reply by GiudGiud** [reply]
An app is the right way. 
I think following what was done for the truss elements is the right thing to do

**Reply by zia-javanbakht** [reply]
@GiudGiud Thank you very much. Can you make a recommendation on which MOOSE systems to learn in general? I mean the right sequence from basic towards more advanced; something like a study roadmap. :)

**Reply by GiudGiud** [reply]
@cbolisetti how did you figure out how to implement the truss / beam elements?
